File: leetcode238.py
class Solution(object):
    def productExceptSelf(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        arr = [1]
        for i in range(len(nums) - 1):
            arr.append(arr[-1]*nums[i])
        temp = 1
        for j in range(len(nums) - 1,-1,-1):
            arr[j] *= temp
            temp *= nums[j]
        return arr

File: leetcodec++/leetcode238.cpp
//
// Created by 黄敏辉 on 2019-09-15.
//

class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        vector<int> res;
        int p = 1;
        for (int i = 0; i < nums.size(); i++){
            res.push_back(p);
            p *= nums[i];
        }
        p = 1;
        for (int i = nums.size() - 1; i >= 0; i--){
            res[i]  *= p;
            p *= nums[i];
        }
        return res;
    }
};